oak/server/dev/dev_oak_loader.cc

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "dev_oak_loader.h"
+
+#include "absl/memory/memory.h"
+#include "asylo/grpc/auth/enclave_server_credentials.h"
+#include "asylo/grpc/auth/null_credentials_options.h"
+#include "asylo/identity/descriptions.h"
+#include "asylo/identity/init.h"
+#include "asylo/util/logging.h"
+#include "asylo/util/statusor.h"
+#include "grpcpp/support/status_code_enum.h"
+#include "include/grpcpp/grpcpp.h"
+
+namespace oak {
+
+DevOakLoader::DevOakLoader() : next_application_id_(0) { InitializeAssertionAuthorities(); }
+
+asylo::StatusOr<oak::CreateApplicationResponse> DevOakLoader::CreateApplication(
+    const oak::ApplicationConfiguration& application_configuration) {
+  std::string application_id = NewApplicationId();
+  LOG(INFO) << "Creating application with ID" << application_id;
+
+  auto runtime = absl::make_unique<OakRuntime>();
+  auto status = runtime->Initialize(application_configuration);
+  if (!status.ok()) {
+    return asylo::Status(status);
+  }
+
+  // Start the runtime.
+  runtime->Start();
+
+  int32_t port = runtime->GetPort();
+  LOG(INFO) << "gRPC server is listening on port: " << port;
+  runtimes_[application_id] = std::move(runtime);
+
+  oak::CreateApplicationResponse response;
+  response.set_application_id(application_id);
+  response.set_grpc_port(port);
+  return response;
+}
+
+grpc::Status DevOakLoader::TerminateApplication(const std::string& application_id) {
+  LOG(INFO) << "Terminating application with ID " << application_id;
+
+  OakRuntime* runtime = runtimes_[application_id].get();
+  if (runtime == nullptr) {
+    LOG(ERROR) << "Unrecognized application ID";
+    return grpc::Status(grpc::StatusCode::INVALID_ARGUMENT, "Unknown application ID");
+  }
+
+  runtime->Stop();
+  LOG(INFO) << "Application with ID " << application_id << " stopped";
+  runtimes_.erase(application_id);
+
+  return grpc::Status::OK;
+}
+
+// Even if we are not running in an enclave, we are still relying on Asylo assertion authorities
```

Comment:
(nit, pre-existing) full stop at end of sentence.

Reply:
It's necessary for the client to use Asylo gRPC
Was moved here from `dev_oak_manager.cc`:
https://github.com/project-oak/oak/blob/664f0d6ee80ab805b72a461b1b879f5fdca0de2a/oak/server/dev/dev_oak_manager.cc#L76-L79

Reply:
Oh, just the full stop. I though I needed to delete Asylo usage somehow.
Fixed